lib/scheduler/scheduler.js

```diff
@@ -184,11 +184,17 @@ function connectEndpoint_p(endpoint, timeout, shouldContinue) {
             logger.trace("Refusing to delete /dev/null.");
             return;
           }
-          logger.trace('Normal exit, deleting log file ' + logFilePath);
-          fs.unlink(logFilePath, function(err) {
-            if (err)
-              logger.warn('Failed to delete log file ' + logFilePath + ': ' + err.message);
-          });
+
+          if (!appSpec || !appSpec.settings || !appSpec.settings.appDefaults || 
```

Comment:
Why the initial `!appSpec`? We don't have that check in other setting fetches.

I notice that there is a pattern in how we fetch setting values off an `appSpec`; maybe that should be consolidated with a helper function like:
```
function getSetting(name, defaultValue) {
  if (appSpec.settings && appSpec.settings.hasOwnProperty(name)) {
    return appSpec.settings[name];
  } else {
    return defaultValue;
  }
}
```

Even better would be if we could assume that `appSpec.settings` were always fully populated with defaults resolved outside this code.